### PR TITLE
Filter tags by namespace on admin page #36

### PR DESCRIPTION
**Pull request by solewniczak**
This is one option but I believe that maybe I should implement it similarly to Access Control List Management where you have namespace selector on the left side.

### CODE REVIEW

File: admin.php [hunk 1]
```diff
@@ -91,7 +117,37 @@ protected function html_table() {
...
         }
-
+        echo '<tr>';
+        echo '<td colspan="5" class="centeralign">';
+        echo '<span class="medialeft">';
+        echo '<button type="submit" name="fn[delete]" id="tagging__del">'.$this->getLang('admin delete_selected').'</button>';
+        echo '</tr>';
+        echo '</form>';
         echo '</table>';
+
```
Comment (splitbrain) [inline admin.php]: Might be a good idea to transform this form into a dokuwiki\Form\Form as well, to make the code more consistent.

File: helper.php [hunk 2]
```diff
@@ -356,12 +356,14 @@ public function tpl_tags($print = true) {
...
+     * @param string $namespace empty for entire wiki
+     * 
      * @return array
      */
-    public function getAllTags() {
+    public function getAllTags($namespace='') {
 
         $db = $this->getDb();
-        $res = $db->query('SELECT pid, tag, tagger FROM taggings ORDER BY tag');
+        $res = $db->query('SELECT pid, tag, tagger FROM taggings WHERE pid LIKE ? ORDER BY tag', "%$namespace%");
```
Comment (splitbrain) [inline helper.php]: As discussed yesterday, this should be adjusted to `"$namespace%"`

File: helper.php [hunk 3]
```diff
@@ -413,5 +415,21 @@ public function renameTag($formerTagName, $newTagName) {
...
+     * Deletes a tag
+     *
+     * @param array $tags
+     */
+    public function deleteTags($tags) {
+        if (empty($tags)) return;
+        
+        $db = $this->getDb();
+        
+        $query = 'DELETE FROM taggings WHERE ' . implode(' OR ', array_fill(0, count($tags), 'CLEANTAG(tag) = ?'));
```
Comment (splitbrain) [inline helper.php]: as discussed yesterday this should be adjusted to only apply to the current namespace filter

File: lang/en/lang.php [hunk 4]
```diff
@@ -11,9 +11,13 @@
 $lang['admin find tag']            = 'Find Tag';
 $lang['admin new name']            = 'New name';
 $lang['admin rename tag']          = 'Rename tag';
+$lang['admin filter']              = 'Filter by namespace';
 $lang['admin enter tag names']     = 'Please enter the current and the new tag name';
 $lang['admin tag does not exists'] = 'The tag name you entered does not exist.';
 $lang['admin renamed']             = 'Your tag has been renamed';
+$lang['admin delete_selected']     = 'Delete Selected';
+$lang['admin filter button']       = 'Filter';
+$lang['admin deleted']             = 'Your have deleted %d tags on %d pages';
```
Comment (splitbrain) [inline lang/en/lang.php]: typo: Your -> You

**Comment by solewniczak** [comment]
Ok, now it should be completed.

**Pull request merged**